lib/id3c/cli/command/geocode.py

```diff
@@ -411,6 +412,34 @@ def parse_first_smartystreets_result(result: list) -> dict:
     return response
 
 
+def extract_address(address: dict) -> dict:
+    """
+    Given an *address*, converts it to text and returns a result from
+    the SmartyStreets US Extract API containing information about an address
+    connected to the text.
+
+    Note that this API is not consistent with the US Street API, and the lookup
+    and responses must be handled differently.
+    """
+    LOG.debug("Previous lookup failed. Looking up address as free text.")
```

Comment:
It's cleaner to write functions that don't care about where they're called from or why, so logging like this is generally better suited at the callsite instead of inside the called function.